### PR TITLE
Weekly checkout thank you page

### PR DESCRIPTION
**Pull request by lucymonie**
## Why are you doing this?
The aim is to create a Weekly checkout thank you page, using the text in the wireframes in this [**Trello Card**](https://trello.com/c/oPBdQeJW/2335-guardian-weekly-thank-you-page), and an image relevant to the Weekly product.

There's a subscriber ID shown in the wireframes, which is not included in this work because this data isn't currently available on the front end. A separate card is in the backlog to add this so we can make it available for the thank you pages.

## Changes
- Text and imagery that's specific to the Guardian Weekly product
- Refactoring some of the components that were duplicated between the pages for the different products so these can be shared
- Update to navbar for all the thank you pages so it doesn't show the word 'Checkout', as requested by @jesseyuen

## Screenshots
Screen Shot 2019-05-30 at 11 42 03
Screen Shot 2019-05-30 at 11 51 35

### CODE REVIEW

File: support-frontend/assets/helpers/subscriptionsForms/formReducer.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ function createFormReducer(
   const { productPrices } = window.guardian;
 
   const initialState = {
-    stage: 'checkout',
+    stage: 'thankyou',
```
Comment (rupertbates) [inline support-frontend/assets/helpers/subscriptionsForms/formReducer.js]: !

File: support-frontend/assets/components/subscriptionCheckouts/stage.jsx [hunk 2]
```diff
@@ -24,6 +22,16 @@ type PropTypes = {|
   countryGroupId: CountryGroupId,
 |};
 
+type StagePropTypes = {
+  stage: Stage,
+  formSubmitted: boolean,
+  countryGroupId: CountryGroupId,
+  checkoutForm: Node,
+  thankYouContentPending: Node,
+  thankYouContentNotPending: Node,
```
Comment (rupertbates) [inline support-frontend/assets/components/subscriptionCheckouts/stage.jsx]: I think you could just call this `thankYouContent`, if you're worried it's not distinct enough you could call the other `pendingContent`.

File: support-frontend/assets/helpers/productPrice/productOptions.js [hunk 3]
```diff
@@ -13,6 +13,9 @@ const Sixday: 'Sixday' = 'Sixday';
 const SixdayPlus: 'SixdayPlus' = 'SixdayPlus';
 const Everyday: 'Everyday' = 'Everyday';
 const EverydayPlus: 'EverydayPlus' = 'EverydayPlus';
+const Annual: 'Annual' = 'Annual';
```
Comment (rupertbates) [inline support-frontend/assets/helpers/productPrice/productOptions.js]: These are not product options they are billing periods which is different. I can't see anywhere you're using them in this PR but if you need another way to do something without adding these ping me on chat and we can discuss.

File: support-frontend/assets/pages/paper-subscription-checkout/components/thankYou.jsx [hunk 4]
```diff
@@ -56,7 +56,7 @@ const whatNext: {[FulfilmentOptions]: Element<*>} = {
         <OrderedList items={[
           <span>
             Look out for an email from us confirming your subscription.
-            It has everything you need to know about how manage it in the future.
+            It has everything you need to know about how to manage it in the future.
```
Comment (rupertbates) [inline support-frontend/assets/pages/paper-subscription-checkout/components/thankYou.jsx]: Good spot 😃

File: support-frontend/assets/components/subscriptionCheckouts/stage.jsx [hunk 5]
```diff
@@ -24,6 +22,16 @@ type PropTypes = {|
   countryGroupId: CountryGroupId,
 |};
 
+type StagePropTypes = {
+  stage: Stage,
+  formSubmitted: boolean,
+  countryGroupId: CountryGroupId,
```
Comment (damien-shaw-guardian) [inline support-frontend/assets/components/subscriptionCheckouts/stage.jsx]: I don't think countryGroupId is used in checkStage

File: support-frontend/assets/components/subscriptionCheckouts/stage.jsx (hunk 2 above)
Comment (lucymonie) [reply support-frontend/assets/components/subscriptionCheckouts/stage.jsx]: 👍

File: support-frontend/assets/helpers/productPrice/productOptions.js (hunk 3 above)
Comment (lucymonie) [reply support-frontend/assets/helpers/productPrice/productOptions.js]: Yes, I realised these are in helpers/billingPeriods.js after you pointed that out. Thanks!

File: support-frontend/assets/components/button/button.scss [hunk 6]
```diff
@@ -112,3 +112,7 @@
     transform: translateX(20%);
   }
 }
+
+.component-button--visited-white-text:visited {
+  color: #ffffff;
```
[the remark this answers is not in the document]
Comment (lucymonie) [reply support-frontend/assets/components/button/button.scss]: Sure can - thanks!

File: support-frontend/assets/components/subscriptionCheckouts/stage.jsx (hunk 5 above)
Comment (lucymonie) [reply support-frontend/assets/components/subscriptionCheckouts/stage.jsx]: Nice pick up! Removed 🥇

File: support-frontend/assets/helpers/billingPeriods.js [hunk 7]
```diff
@@ -3,8 +3,10 @@
 const Annual: 'Annual' = 'Annual';
 const Monthly: 'Monthly' = 'Monthly';
 const Quarterly: 'Quarterly' = 'Quarterly';
+const SixForSix: 'SixForSix' = 'SixForSix';
```
Comment (rupertbates) [inline support-frontend/assets/helpers/billingPeriods.js]: `SixForSix` is actually called `SixWeekly`, this is to match with existing code elsewhere

Comment (lucymonie) [reply support-frontend/assets/helpers/billingPeriods.js]: Ah - I left both in there. Is it changed across the board now?

Comment (rupertbates) [reply support-frontend/assets/helpers/billingPeriods.js]: Yes it should be

**Pull request merged**